File: python/test/longestCommonSubstringDistanceTest.py
import unittest
from longestCommonSubstringDistance import calculate

class LongestCommonSubstringTest(unittest.TestCase):

	def test_lcss_distance_of_identical_words(self):
		lcss_distance = calculate("hi","hi")
		self.assertEqual(lcss_distance, 2)

	def test_lcss_distance_of_identical_words_case_senseitive(self):
		lcss_distance = calculate("hi","Hi")
		self.assertEqual(lcss_distance, 1)		

	def test_lcss_distance_of_similar_words(self):
		lcss_distance = calculate("cat","bat")
		self.assertEqual(lcss_distance, 2)		

	def test_lcss_distance_of_subset_words(self):
		lcss_distance = calculate("cat","catch")
		self.assertEqual(lcss_distance, 3)

	def test_lcss_distance_of_anagrams(self):
		lcss_distance = calculate("tame","mate")
		self.assertEqual(lcss_distance, 1)

	def test_lcss_distance_of_words_with_similar_letters(self):
		lcss_distance = calculate("act","catch")
		self.assertEqual(lcss_distance, 1)

	def test_lcss_distance_of_words_with_same_letters(self):
		lcss_distance = calculate("ipcicadlyl","piccadilly")
		self.assertEqual(lcss_distance, 3)

File: java/src/test/java/com/hakanozbay/metric/LongestCommonSubstringDistanceTest.java
package com.hakanozbay.metric;

import static org.junit.Assert.assertEquals;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class LongestCommonSubstringDistanceTest {

	LongestCommonSubstringDistance longestCommonSubstringDistance;
	
	@Before
	public void setUp() 
	{
		longestCommonSubstringDistance = new LongestCommonSubstringDistance();
	}

	@After
	public void tearDown() 
	{
		longestCommonSubstringDistance = null;
	}

	@Test
	public void testLcssDistanceOfIdenticalWords() 
	{
		double distance = longestCommonSubstringDistance.calculate("hi", "hi");
		assertEquals(2, distance, 0);
	}
	
	@Test
	public void testLcssDistanceOfIdenticalWordsCaseSensitive() 
	{
		double distance = longestCommonSubstringDistance.calculate("hi", "Hi");
		assertEquals(1, distance, 0);
	}
	
	@Test
	public void testLcssDistanceOfSimilarWords() 
	{
		double distance = longestCommonSubstringDistance.calculate("cat", "bat");
		assertEquals(2, distance, 0);
	}
	
	@Test
	public void testLcssDistanceOfSubsetWords() 
	{
		double distance = longestCommonSubstringDistance.calculate("cat", "catch");
		assertEquals(3, distance, 0);
	}
	
	@Test
	public void testLcssDistanceOfAnagrams() 
	{
		double distance = longestCommonSubstringDistance.calculate("tame", "mate");
		assertEquals(1, distance, 0);
	}
	
	@Test
	public void testLcssDistanceOfWordsWithSimilarLetters() 
	{
		double distance = longestCommonSubstringDistance.calculate("act", "catch");
		assertEquals(1, distance, 0);
	}
	
	@Test
	public void testLcssDistanceOfWordsWithSameLetters() 
	{
		double distance = longestCommonSubstringDistance.calculate("ipcicadlyl", "piccadilly");
		assertEquals(3, distance, 0);
	}
	

}
